a3/test.ml

```diff
@@ -3,13 +3,13 @@ open OUnit2;;
 let test_find_zero_exact (exp:string) : unit =
   let e = ExpressionLibrary.parse exp in
   (* For some reason, cmp_float is failing here. Rather than debug too much, use this. *)
-  let my_cmp_float a b epsilon : bool = ((abs_float (a -. b)) < epsilon) in
+  let my_cmp_float epsilon a b : bool = ((abs_float (a -. b)) < epsilon) in
   match Expression.find_zero_exact e with
   | Some solution ->
     let s = (Expression.evaluate solution 123.) in
     let zero = (Expression.evaluate e s) in
 
-    assert_bool "not equal" (my_cmp_float zero 0. 0.001)
+    assert_equal ~cmp:(my_cmp_float 0.001) ~printer:string_of_float 0. zero
```

Comment:
From an old branch cleaning up `a3` tests. It turns out `assert_equal` takes an optional `~cmp` param, so we don't need to do our own comparing. This also adds a printer for better results.